File: articles/edsger-dijkstra-and-the-discipline-of-structured-programming/c/structured_loop.c
#include <stdio.h>

int sum_to(int n) {
    int acc = 0;
    for (int i = 0; i <= n; i++) {
        acc += i;
    }
    return acc;
}

int main(void) {
    printf("sum_to_5=%d\n", sum_to(5));
    return 0;
}

File: articles/edsger-dijkstra-and-the-discipline-of-structured-programming/go/structured_loop.go
package main

import "fmt"

func sumTo(n int) int {
	acc := 0
	for i := 0; i <= n; i++ {
		acc += i
	}
	return acc
}

func main() {
	fmt.Printf("sum_to_5=%d\n", sumTo(5))
}
